src/runtime/index.ts

```diff
@@ -56,10 +61,10 @@ export function runtime(defaults: RequestOpts) {
       : false;
 
     if (isJson) {
-      return { status, data: data ? JSON.parse(data) : null } as T;
+      return { status, data: data ? JSON.parse(data) : null, headers } as T;
     }
 
-    return { status, data } as T;
+    return { status, data, headers } as T;
```

Comment:
Instead of picking the different fields here I think it might be better to just expose the original response entirely so that all of the fields can be accessed.

Reply:
This required some non-trivial typing changes (e.g., to `ApiResponse`), but my project that depends on this library type checked without any changes using the latest revision.